sdk/core/azure-core-amqp/vendor/azure-uamqp-c/src/connection.c

```diff
@@ -173,49 +173,64 @@ static int channel_table_allocate(CHANNEL_TABLE* channel_table, uint16_t* h)
     else
     {
         *h = 0;
-
         if (channel_table->size == channel_table->capacity)
         {
-          CHANNEL_TABLE_ENTRY* new_handles = (CHANNEL_TABLE_ENTRY*)realloc(
-              channel_table->channels, sizeof(CHANNEL_TABLE_ENTRY) * channel_table->capacity * 2);
-          if (new_handles == NULL)
-          {
-            LogError("Cannot reallocate memory for handle table");
-            result = MU_FAILURE;
-          }
-          else
-          {
-            channel_table->channels = new_handles;
-            // Ensure the newly resized handle table is empty.
-            for (uint16_t i = channel_table->size; i < channel_table->capacity * 2; i++)
+            if (channel_table->capacity == UINT16_MAX)
             {
-              channel_table->channels[i].outgoing_channel = ENDPOINT_UNUSED_CHANNEL;
-              channel_table->channels[i].incoming_channel = ENDPOINT_UNUSED_CHANNEL;
-              channel_table->channels[i].is_endpoint_live = false;
+                LogError("Cannot allocate more channels");
+                result = MU_FAILURE;
+            }
+            else
+            {
+                uint32_t new_capacity = channel_table->capacity + INITIAL_CHANNEL_TABLE_CAPACITY;
```

Comment:
Just confirming
1. We're ok with this growing as `+= 16` instead of `*= 2` as before?
2. No sane person would set `INITIAL_CHANNEL_TABLE_CAPACITY` to 0

Reply:
For #2, the code won't work if you set the initial capacity to 0. And this isn't a customer configuration value - it's a #define to make coding simpler so I can change it in one place not many.

And for #1, exponential growth is potentially problematic.